aptos-move/framework/aptos-framework/sources/jwks.move

```diff
@@ -0,0 +1,480 @@
+/// JWK functions and structs.
+module aptos_framework::jwks {
+    use std::error::invalid_argument;
+    use std::option;
+    use std::option::Option;
+    use std::string::String;
+    use std::vector;
+    use aptos_std::copyable_any::{Any, pack};
+    use aptos_framework::event::emit;
+    use aptos_framework::reconfiguration;
+    use aptos_framework::system_addresses;
+    #[test_only]
+    use std::string;
+    #[test_only]
+    use std::string::utf8;
+    #[test_only]
+    use aptos_std::comparator::{compare_u8_vector, is_greater_than, is_equal};
+    #[test_only]
+    use aptos_std::copyable_any;
+
+    friend aptos_framework::genesis;
+
+    const EUNEXPECTED_EPOCH: u64 = 1;
+    const EUNEXPECTED_VERSION: u64 = 2;
+    const EUNKNOWN_JWK_MAP_EDIT: u64 = 3;
+
+    /// An OIDC provider.
+    struct OIDCProvider has drop, store {
```

Comment:
Why not use a UTF8 string here?

Reply:
1. why does it have to be limited to UTF8 if we can easily allow more?

2. One `utf8()` call is saved if the type of name is `vector<u8>`.

3. In case later we want to sort these providers by name in alphanumeric order, using Move string as the sorting key will cause some trouble: when we compare 2 move strings in move, we are actually vector-u8-comparing the bcs serializations of the 2 strings (i.e., `utf8(b"bob") > utf8(b"alice")`), which is unwanted behavior in most cases.

It's possible to use the `string::bytes` function to copy out the raw bytes and vector-u8-compare, but if we are doing this work for every comparison, then why not use the raw bytes as the name directly?